Adding trace ID sampling to Log Index documentation (#36104)
DataDog/documentation: content/en/logs/log_configuration/indexes.md

* Default **query** is `*`, meaning all logs flowing in the index would be excluded. Scope down exclusion filter to only a subset of logs [with a log query][12].
* Default **sampling rule** is `Exclude 100% of logs` matching the query. Adapt sampling rate from 0% to 100%, and decide if the sampling rate applies on individual logs, or group of logs defined by the unique values of any attribute.

>   * If the sampling rate applies on individual logs, sampling will be performed on the existence of trace IDs in the logs, if present. Logs sampled will therefore have an increased chance to correlated with sampled traces, to promote unified telemetry data.

Reviewer: ```suggestion
  * If the sampling rate applies on individual logs, sampling is performed on the existence of trace IDs in the logs, if present. In this scenario, logs sampled have an increased chance to be correlated with sampled traces, to promote unified telemetry data.
```

>   * If the unique value of trace ID is chosen for sampling, the behaviour is the same as on individual logs.

Reviewer: ```suggestion
  * If the unique value of a trace ID is chosen for sampling, the behavior is the same as on individual logs.
```
